roles/get-sitename/tasks/main.yml

```diff
@@ -53,28 +53,90 @@
   when:
     acsf_environment != ""
 
-- name: Get list of sites on Site Factory
-  uri:
-    url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
-    method: GET
-    user: "{{ acquia_username }}"
-    password: "{{ acquia_api_key }}"
-    force_basic_auth: yes
-    return_content: yes
-  register: existing_sites
+# - name: Get list of sites on Site Factory
+#   uri:
+#     url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
+#     method: GET
+#     user: "{{ acquia_username }}"
+#     password: "{{ acquia_api_key }}"
+#     force_basic_auth: yes
+#     return_content: yes
+#   register: existing_sites
 
-#- name: Display all site names
-#  debug:
-#    var: item
-#  with_items: "{{ existing_sites | json_query('json.sites[*].site') }}"
+- name: Get sites.json from remote
+  command: drush @acsf{{ drush_environment }}.cardinald7.saml php-eval "include 'sites/sites.php'; drupal_json_output(gardens_site_data_load_file());"
+  register: sites_json_raw
 
-- name: Establish numerical id from existing_sites JSON response
+- name: Get sites.json from remote
   set_fact:
-    site_id: "{{ existing_sites | json_query(existing_sites_query) }}"
+    sites_json: "{{ sites_json_raw.stdout|from_json }}"
+
+# - name: debug sites.json
+#   debug:
+#     var: sites_json
+
+#
+# Creates a format that looks like:
+# {
+#     "id": 5001626,
+#     "site": "crceducation",
+#     "url": "crceducation-dev.cardinalsites.stanford.edu"
+# },
+#
+# From something that looks like
+#  {
+#     "clayman.dev-cardinalsites.acsitefactory.com": {
+#     "name": "sbiybj5001706dev",
+#     "flags": [],
+#     "conf": {
+#       "gardens_site_id": 5001706,
+#       "gardens_db_name": "sbiybj5001706dev",
+#       "acsf_site_id": 5001706,
+#       "acsf_db_name": "sbiybj5001706dev",
+#       "site_api_key": "400b006fda61ada556d9e19171d4b88edef1fa82"
+#     }
+#   },
+#
+# The REST API has a restriction of 100 items per request and quickly falls over
+# for our needs. To bypass the API we are directly getting all of the site info
+# from sites.json on the acquia server. Unfortunately the initial format is not
+# usable and we need to parse out the machine name and id of the site factory
+# site by using the url for the sitefactory item as the machine name and getting
+# the key from the json array for the node id. We only use the acsitefactory uri
+# as it is the canonical machine name for the site and the other domains listed
+# cannot be trusted.
+#
+- name: Manipulate sites.json
+  set_fact:
+    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': item.1.acsf_site_id, 'site': item.0|regex_replace('.(dev-|stage-|test-)cardinalsites.acsitefactory.com', '') } ] }}"
+  with_together:
+    - "{{ sites_json.sites }}"
+    - "{{ sites_json|json_query('sites.*.conf') }}"
+  when: item.0 | regex_search('acsitefactory')
```

Comment:
This condition will prevent other TLDs from polluting the list of possible sites and can be trusted as it is the canonical machine name value the site gets on creation.